xamarin/Xamarin.Forms
# Implement FormsStack and FormsQueue internally for VS 2017 Compatibility

### Description of Change ###
- Update everything back to using VS 2019 build components
- Copy in Stack<T> and Queue<T> implementations from corefx so that the moved types in mono don't break VS2017
   - I just linked the files into the platform projects so they can all stay internal/sealed. I assume once we totally drop VS 2017 we'll just delete all of this and roll back to using the packaged versions of these

Here are some comparison tests ran through Benchmark dotnet

|     Method |      Mean |     Error |    StdDev |
|----------- |----------:|----------:|----------:|
|  PushQueue | 27.780 ms | 0.5488 ms | 1.3148 ms |
|   PopQueue |  5.082 ms | 0.0992 ms | 0.0928 ms |
| PushFQueue | 26.596 ms | 0.5269 ms | 0.8046 ms |
|  PopFQueue |  2.397 ms | 0.0453 ms | 0.0378 ms |


|     Method |      Mean |     Error |    StdDev |    Median |
|----------- |----------:|----------:|----------:|----------:|
|  PushStack | 28.324 ms | 1.0790 ms | 3.0433 ms | 26.999 ms |
|   PopStack |  4.148 ms | 0.0827 ms | 0.0813 ms |  4.125 ms |
| PushFStack | 26.596 ms | 0.6836 ms | 1.2670 ms | 26.244 ms |
|  PopFStack |  2.086 ms | 0.0454 ms | 0.0819 ms |  2.053 ms |

Here's the test project if you would like to verify or add additional tests

[TestCollections.zip](https://github.com/xamarin/Xamarin.Forms/files/3816504/TestCollections.zip)



### Issues Resolved ### 
- fixes #8126


### Platforms Affected ### 


- Core/XAML (all platforms)
- iOS
- Android
- UWP

### Testing Procedure ###
- pull down the nuget and make sure you can build and deploy projects on VS 2017

### PR Checklist ###
